File: Python/URI2674.py
import sys

def ehPrimo(n):
    if(n == 0 or n == 1):
        return False

    cont = 0
    for i in range(1,n+1):
        if(n%i==0):
            cont += 1
        if(cont > 2):
            break
    
    return cont == 2 if True else False

for line in sys.stdin:
    n = int(line)

    if(n < 10):
        if(n == 0 or n == 1):
            print("Nada")
        elif(ehPrimo(n)):
            print("Super")
        else:
            print("Nada")
    else:
        ns = str(n)
        cont = 0
        ehP = ehPrimo(n)

        if(ehP):
            for i in range(len(ns)):
                if(ehPrimo(int(ns[i]))):
                    cont += 1
                else:
                    break
        
        if(cont == len(ns) and ehP):
            print("Super")
        elif(ehP):
            print("Primo")
        else:
            print("Nada")

File: C/uri2674.c
#include <stdio.h>
#include <stdlib.h>

int ehPrimo(int n){
    if(n == 0 || n == 1){
        return 0;
    }

    int cont = 0;
    for(int k = 1; k <= n; k++){
        if(n%k==0){
            cont++;
        }

        if(cont > 2){
            return 0;
        }
    }

    return 1;
}

int contaDigitos(int valor){
    int contaDigitos = 0;

    while (valor != 0){
        contaDigitos++;
        valor = valor / 10;
    }
    return contaDigitos;
}

void main(){
    int n;
    while(scanf("%d", &n) != EOF){
        if(n < 10){
            if(n == 0 || n == 1){
                printf("Nada\n");
            }
                
            else if(ehPrimo(n)){
                printf("Super\n");
            }else{
                printf("Nada\n");
            }
        }else{
            int t = 10;
            t = contaDigitos(n);
            char y[t] = "";
            sprintf(y, "%i", n);
            int cont = 0;
            int ehP = ehPrimo(n);

            if(ehP){
                for (int i = 0; i < sizeof(y); i++){
                    char u[1] = b;
                    if(ehPrimo(atoi(u))){
                        cont++;
                    }else{
                        break;
                    }
                }
            }
            
            if(cont == sizeof(y) && ehP){
                printf("Super\n");
            }else if(ehP){
                printf("Primo\n");
            }else{
                printf("Nada\n");
            }  
        }
    }
}